src/App.js

```diff
@@ -1,19 +1,140 @@
 import React from 'react';
-import './App.css';
+import './App.scss';
 
+import TodoList from './components/TodoList/TodoList';
+
+import todos from './api/todos';
 import users from './api/users';
 
-function App() {
-  return (
-    <div className="App">
-      <h1>Add todo form</h1>
-
-      <p>
-        <span>Users: </span>
-        {users.length}
-      </p>
-    </div>
-  );
+class App extends React.Component {
+  state = {
+    todos,
+    users,
+    titleEntered: true,
+    nameChoosed: true,
+    titleInput: '',
+    selectInput: '',
+  }
+
+  addTodo = (event) => {
+    const userId = this.state.selectInput;
+    const title = this.state.titleInput;
+
+    if (userId !== '' && title !== '') {
+      const todo = {
+        id: this.state.todos.length + 1,
+        userId: Number(userId),
+        title,
+      };
+
+      this.setState(
+        state => ({
+          todos: [...state.todos, todo],
+          titleEntered: true,
+          nameChoosed: true,
+        }),
+      );
+      event.target.reset();
+    }
+
+    if (title) {
+      this.setState({ titleEntered: true });
+    } else {
+      this.setState({ titleEntered: false });
+    }
+
+    if (userId) {
+      this.setState({ nameChoosed: true });
+    } else {
+      this.setState({ nameChoosed: false });
+    }
+
+    event.preventDefault();
+  }
+
+  inputHandler = (event) => {
+    this.setState({
+      titleEntered: true,
+      titleInput: event.target.value,
+    });
+  }
+
+  selectHandler = (event) => {
+    this.setState({
+      nameChoosed: true,
+      selectInput: event.target.value,
+    });
+  }
+
+  render() {
+    const preparedTodos = this.state.todos.map(todo => ({
+      ...todo,
+      user: this.state.users.find(user => user.id === todo.userId),
+    }));
```

Comment:
prepare your data outside of the component, keep in the state already prepared todos